pkgs/development/python-modules/pyjwt/default.nix

```diff
@@ -31,7 +31,7 @@ buildPythonPackage rec {
 
   disabledTests = lib.optionals (lib.versionOlder version "2.0") [
```

Comment:
How is this supposed to work? The expression will always be eval'ed against the hardcoded `version` further up the the file. Overrides will not replace this and also not magically change the way `version` is bound (e.g. via `rec`).

Example on why this doesn't work:

Lets say you have a function: 
```
a: rec {
  version = "some-version";
  something = if version == "123" then "foo" else version;
}
```

The result will always be `{ version = "some-version"; something = foo"; }`.

If you could `overrideAttrs (_: { version = "123"; })` on the above it would desugar to something like this:

```
rec {
  version = "some-version";
  something = if version == "123" then "foo" else version;
} // {
  version = "123";
}
```

Which equals `{ version = "123"; something = "some-version"; }`.

Reply:
Thanks for the explanation.